### PR TITLE
Fixed r_sys_pid_to_path (supports winxp -> win10)

### PR DESCRIPTION
**Pull request by MapleLeaf-X**

### CODE REVIEW

File: libr/util/sys.c [hunk 1]
```diff
@@ -923,69 +923,30 @@ R_API int r_is_heap (void *p) {
...
+	processHandle = OpenProcess (PROCESS_QUERY_INFORMATION, FALSE, pid);
+	if (processHandle == NULL) {
+		eprintf ("r_sys_pid_to_path: Cannot open process.\n");
+		goto err_r_sys_pid_to_path;
+	}
+	if (GetProcessImageFileName (processHandle, filename, maxlength) == 0) {
+		eprintf ("r_sys_pid_to_path: Error calling GetProcessImageFileName\n");
+		goto err_r_sys_pid_to_path;
+	}
+	tmp = malloc (strlen(filename) + 14);
```
Comment (radare) [inline libr/util/sys.c]: just use r_str_newf ()

File: libr/util/sys.c [hunk 2]
```diff
@@ -923,69 +923,30 @@ R_API int r_is_heap (void *p) {
...
-		}
-		CloseHandle (handle);
-		return r_sys_conv_win_to_utf8 (filename);
-	}
-	return NULL;
-#else
-	HANDLE processHandle = NULL;
-	TCHAR filename[FILENAME_MAX];
+	const char* result = NULL;
+	char* tmp;
```
Comment (radare) [inline libr/util/sys.c]: define the variables in the same line that are asssigned below

File: libr/util/sys.c [hunk 3]
```diff
@@ -923,69 +923,30 @@ R_API int r_is_heap (void *p) {
...
-		CloseHandle (processHandle);
-	} else {
-		eprintf ("r_sys_pid_to_path: Cannot open process.");
-	}
-	return NULL;
-#endif
+	processHandle = OpenProcess (PROCESS_QUERY_INFORMATION, FALSE, pid);
+	if (processHandle == NULL) {
+		eprintf ("r_sys_pid_to_path: Cannot open process.\n");
+		goto err_r_sys_pid_to_path;
```
Comment (radare) [inline libr/util/sys.c]: i would just drop the gotos and use } eles if .. } else { ..}

File: libr/util/sys.c [hunk 4]
```diff
@@ -923,69 +923,32 @@ R_API int r_is_heap (void *p) {
...
+	}
+	length = GetProcessImageFileName (processHandle, filename, maxlength);
+	if (length == 0) {
+		eprintf ("r_sys_pid_to_path: Error calling GetProcessImageFileName\n");
+		goto err_r_sys_pid_to_path;
+	}
+	tmp = malloc (length + 14);
```
Comment (radare) [inline libr/util/sys.c]: use r_str_newf

File: libr/util/sys.c [hunk 5]
```diff
@@ -923,69 +923,32 @@ R_API int r_is_heap (void *p) {
 
 R_API char *r_sys_pid_to_path(int pid) {
 #if __WINDOWS__
-	HANDLE kernel32 = GetModuleHandle (TEXT("kernel32"));
```
Comment (radare) [inline libr/util/sys.c]: cc @sanguinawer can you review this?

File: libr/util/sys.c [hunk 6]
```diff
@@ -923,69 +923,32 @@ R_API int r_is_heap (void *p) {
...
-		eprintf ("r_sys_pid_to_path: Cannot open process.");
-	}
-	return NULL;
-#endif
+	processHandle = OpenProcess (PROCESS_QUERY_INFORMATION, FALSE, pid);
+	if (processHandle == NULL) {
+		eprintf ("r_sys_pid_to_path: Cannot open process.\n");
+		goto err_r_sys_pid_to_path;
+	}
+	length = GetProcessImageFileName (processHandle, filename, maxlength);
```
Comment (radare) [inline libr/util/sys.c]: define and assign in the sasme line

File: libr/util/sys.c [hunk 7]
```diff
@@ -923,69 +923,32 @@ R_API int r_is_heap (void *p) {
...
+	if (length == 0) {
+		eprintf ("r_sys_pid_to_path: Error calling GetProcessImageFileName\n");
+		goto err_r_sys_pid_to_path;
```
Comment (radare) [inline libr/util/sys.c]: avoid gotos

**Comment by MapleLeaf-X** [comment]
This fix will have to wait for a little bit because it introduces a new bug. (ood; ood fails)

File: libr/util/sys.c [hunk 8]
```diff
@@ -923,69 +923,27 @@ R_API int r_is_heap (void *p) {
...
+	if (length == 0) {
+		eprintf ("r_sys_pid_to_path: Error calling GetProcessImageFileName\n");
 		CloseHandle (processHandle);
-	} else {
-		eprintf ("r_sys_pid_to_path: Cannot open process.");
+		return NULL;
 	}
-	return NULL;
-#endif
+	char *tmp = r_str_newf("\\\\.\\GLOBALROOT%s", filename);
```
Comment (radare) [inline libr/util/sys.c]: missgin space before ( and check fffor null, if (tmp) { ...}

File: libr/util/sys.c [hunk 9]
```diff
@@ -923,69 +923,27 @@ R_API int r_is_heap (void *p) {
 
 R_API char *r_sys_pid_to_path(int pid) {
 #if __WINDOWS__
-	HANDLE kernel32 = GetModuleHandle (TEXT("kernel32"));
-	if (!kernel32) {
```
Comment (radare) [inline libr/util/sys.c]: all this deleted code is no longer necessary?

Comment (MapleLeaf-X) [reply libr/util/sys.c]: It was faulty code and it never worked as intended.

File: libr/util/sys.c [hunk 10]
```diff
@@ -923,69 +923,31 @@ R_API int r_is_heap (void *p) {
...
 		CloseHandle (processHandle);
-	} else {
-		eprintf ("r_sys_pid_to_path: Cannot open process.");
+		return NULL;
 	}
-	return NULL;
-#endif
+	CloseHandle (processHandle);
+	char *tmp = r_str_newf ("\\\\.\\GLOBALROOT%s", filename);
+	if(!tmp) {
```
Comment (radare) [inline libr/util/sys.c]: Missing space before (

File: libr/util/sys.c [hunk 11]
```diff
@@ -923,69 +923,31 @@ R_API int r_is_heap (void *p) {
...
 
-	processHandle = OpenProcess (PROCESS_QUERY_INFORMATION | PROCESS_VM_READ, FALSE, pid);
-	if (processHandle != NULL) {
-		if (GetModuleFileNameEx (processHandle, NULL, filename, FILENAME_MAX) == 0) {
-			eprintf ("r_sys_pid_to_path: Cannot get module filename.");
-		} else {
-			return r_sys_conv_win_to_utf8 (filename);
-		}
+	processHandle = OpenProcess (PROCESS_QUERY_INFORMATION, FALSE, pid);
+	if (processHandle == NULL) {
```
Comment (radare) [inline libr/util/sys.c]: Use !processHandle instead of ==NULL

**Pull request merged**